osu.Game.Rulesets.Mania/Mods/ManiaModFadeIn.cs

```diff
@@ -2,14 +2,25 @@
 // See the LICENCE file in the repository root for full licence text.
 
 using System;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Colour;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
 using osu.Framework.Graphics.Sprites;
 using osu.Game.Graphics;
+using osu.Game.Rulesets.Mania.Configuration;
 using osu.Game.Rulesets.Mania.Objects;
+using osu.Game.Rulesets.Mania.UI;
 using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.UI;
+using osuTK.Graphics;
 
 namespace osu.Game.Rulesets.Mania.Mods
 {
-    public class ManiaModFadeIn : Mod
+    public class ManiaModFadeIn : Mod, IApplicableToDrawableRuleset<ManiaHitObject>
```

Comment:
I know this PR is intended to implement this for fadein, but implementing for hidden should be 5 LOC max, and I think it starts with this mod inheriting from `ManiaModHidden`. From there, move all the code below into `ManiaModHidden` and expose a virtual method `CreateLaneCover()` to apply the scale I mentioned.